v5/core/container_authenticator.go

```diff
@@ -54,7 +54,8 @@ type ContainerAuthenticator struct {
 
 	// [optional] The IAM token server's base endpoint URL.
 	// Default value: "https://iam.cloud.ibm.com"
-	URL string
+	URL     string
+	urlInit sync.Once
```

Comment:
I updated the container and iam authenticators so that they initialize the URL field the same way the vpc authenticator does for consistency, plus it improves testability just a bit.